### ISSUE TITLE
How can I group multiple marathon tasks together for a single report?

### ISSUE THREAD

**Issue by ZakTaccardi** [MarathonLabs/marathon#359]
How does grouping test output work with Marathon?

Imagine that I have three marathon tasks:

```
:catLibA:marathonDebugAndroidTest
:catLibB:marathonDebugAndroidTest
:dogLib:marathonDebugAndroidTest
```

I would like to create a custom task like:

```
:marathonCatLibDebugAndroidTest
```

I would like it to effectively run the tests in the following two tasks

```
:catLibA:marathonDebugAndroidTest
:catLibB:marathonDebugAndroidTest
```

But not `:dogLib:marathonDebugAndroidTest`

However, rather than view the reports for the two "cat lib" tests separately, I would like to create a single merged report so I can view them all together. Does the gradle plugin support this use case?

**Comment by Malinskiy** [comment]
You can group the tasks using gradle API, e.g. [dependsOn](https://docs.gradle.org/current/dsl/org.gradle.api.Task.html#org.gradle.api.Task:dependsOn(java.lang.Object[]))

As for the merging of the report - it's not possible at the moment. You can combine several allure reports using allure tooling and you can use standard JUnit tooling for merging several JUnit XML reports

**Comment by ZakTaccardi** [comment]
> As for the merging of the report - it's not possible at the moment

Is this on the roadmap by any chance? Would be an amazing feature. And if [incremental android tests](https://issuetracker.google.com/issues/146360114) are ever supported, this could be incredibly useful.

> You can combine several allure reports using allure tooling

Is the Marathon report allure compatible? My use case is that I have a jenkins job that may run up to 10 different marathon tasks (generating 10 different reports). And I would like to publish a single HTML folder that contains a way to access all that information (such as failed tests). And if the allure report could deep link into the Marathon report, that would be amazing.

> you can use standard JUnit tooling for merging several JUnit XML reports

oh this is interesting. I would imagine that there is built in support to merge multiple junit reports.

For both allure and junit reporting, their gradle tasks would both need a `:lib:marathonDebugAndroidTest` task output as input.

so if I have:

```
task generateAllureTestReportTask {
  dependsOn ":lib:marathonDebugAndroidTest"
}
```

And then I execute `generateAllureTestReportTask` - how do I ensure that the allure task still generates a report when its input task fails? (without using the `--continue` flag) Or am I thinking about this the wrong way?

**Comment by Malinskiy** [comment]
No, this is not on the roadmap. PRs are welcome after the initial discussion about the feature. I'll be happy to support the tooling for merging the reports.

As for the incremental android tests - I believe this has a serious effect on the metrics of tests probability and will affect the stability of test executions as well as performance and I strongly believe that no one should ever use it.

One of the reports marathon generates is the allure report. So yes, it is allure compatible. Please refer to the allure docs for help on merging these. Same thing is true for JUnit reports.

As for the merging of tasks - there is an option `ignoreFailures` that will always return `true` for the task as well as for CLI. This way you don't have to use the gradle trickery for continuing in case of failure

**Comment by Malinskiy** [comment]
PR #523 introduced the ability to test multiple modules using one test run. This kind of helps with having an aggregated test report, but only works using the CLI. Gradle plugin is not aware of this feature.
